src/pybind/mgr/snap_schedule/module.py

```diff
@@ -250,11 +272,14 @@ def snap_schedule_deactivate(self,
         Deactivate a snapshot schedule for <path>
         '''
         try:
-            use_fs = fs if fs else self.default_fs
-            if not self.has_fs(use_fs):
-                return -errno.EINVAL, '', f"no such filesystem: {use_fs}"
+            if not fs:
```

Comment:
minor nit (feel free to ignore): the block of code

```
            if not fs:
                rc, fs, err = self._default_fs
                if rc < 0:
                    return rc, fs, err
            if not self.has_fs(fs):
                return -errno.EINVAL, '', f"no such filesystem: {fs}"
```

is pretty repetitive and can be avoided by declaring a helper.

Reply:
throwing an exception is easier ... but checking the return value and conditionally returning an error tuple makes it cumbersome

or are you suggesting something like:
```
    rc, fs, err = self.validate_fs(fs)
    if rc < 0:
        return rc, fs, err
```

Reply:
yes, something like that.

Reply:
* added `def _validate_fs(fs)` method
fyi - haven't run teuthology jobs after making this change

Reply:
Please run the change through the test and share the link. Should be good for merging once the tests pass.